### PR TITLE
feat: Apply webpack-defaults v1.x

### PR DESCRIPTION
**Pull request by d3viant0ne**
**What kind of change does this PR introduce?**

**Refactoring**

**Did you add tests for your changes?**

**If relevant, did you update the README?**

**Summary**

 - Retains index.js file history
 - Assumes proposed change to npm@5
 - Assumes proposed appveyor proposal

Can't rebase in the changes from master given the `index.js` in the old defaults upgrade was deleted. 

Has the latest proposed changes to defaults ( appveyor & npm@5 )

**Does this PR introduce a breaking change?**

_Breaking Change_ - Enforces NodeJS > 4.3 via engines.

Must release as semver `MAJOR`

**Other information**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/*
+  MIT License http://www.opensource.org/licenses/mit-license.php
+  Author Tobias Koppers @sokra
+*/
+import path from 'path';
+import loaderUtils from 'loader-utils';
+
+export default function fileLoader(content) {
+  this.cacheable && this.cacheable(); // eslint-disable-line no-unused-expressions
```
Comment (michael-ciniawsky) [inline src/index.js]: This isn't needed with webpack >= v2.0.0, I know the PR 'just' apply `webpack-defaults` for the moment, so could be done later, but it's a simple change so maybe :)

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  if (config.outputPath) {
+    // support functions as outputPath to generate them dynamically
+    outputPath = (
+      typeof config.outputPath === 'function' ? config.outputPath(url) : config.outputPath
+    );
+  }
+
+  const filePath = this.resourcePath;
+  if (config.useRelativePath) {
+    const issuerContext = this._module && this._module.issuer // eslint-disable-line no-underscore-dangle
```
Comment (michael-ciniawsky) [inline src/index.js]: Maybe we should relax some of these ESLint rules or are these lines subject to change in a smaller upcoming PR by intention ?

File: src/index.js (hunk 1 above)
Comment (d3viant0ne) [inline src/index.js]: No, you're right about that. File loader was the guinea pig for defaults. It's being redone only to retain the git history on the lib files and to save me the trouble of having to rebase in changes on a file that was deleted on this branch :)

**Comment by d3viant0ne** [comment]
The CI problem is a linux only thing, i'll figure it out this afternoon.

**Pull request closed without merging**